src/rabbit_exchange_type_consistent_hash.erl

```diff
@@ -84,9 +85,34 @@ route(#exchange { name      = Name,
             Destinations
     end.
 
-validate(_X) -> ok.
+validate(#exchange { arguments = Args }) ->
+    case hash_args(Args) of
+        {undefined, undefined} -> ok;
+        {undefined, {_Type, Value}} ->
+            case lists:member(Value, ?PROPERTIES) of
+                true  -> ok;
+                false ->
+                    rabbit_misc:protocol_error(precondition_failed,
+                                               "Unsupported property: ~s",
+                                               [Value])
+            end;
+        {_, undefined} -> ok;
+        {_, _} ->
+            rabbit_misc:protocol_error(precondition_failed,
+                                       "hash-header and hash-property are mutually exclusive",
+                                       [])
+    end.
 
-validate_binding(_X, _B) -> ok.
+validate_binding(_X, #binding { key = K }) ->
```

Comment:
If this change and related tests is related to #6 then it would be nice to have independent PRs for that. I think fixing the negative routing key issue could even go into 3.5.5. What do you think @michaelklishin ? 

Sorry I didn't spot this earlier, but we usually pack one fix per PR